devices/tplink_re200-v4.yml

```diff
@@ -0,0 +1,23 @@
+vendor: TP-Link
+model: RE200
+variant: v4
+bootloader: unknown
+cpu: MediaTek MT7628AN
+cpu_cores: 1
+cpu_mhz: 580
+memory_mb: 64
+storage:
+  - type: unknown
+    size_mb: 8
+device_type: WiFi Router
+ethernet_100m_ports: 1
+jtag: false
+serial: 57600 / 8N1
+outdoor: false
+supported_current_rel: none
+supported_since_rel: snapshot
+target: ramips
+vlan: true
```

Comment:
I'm not sure whether vlan can be limited to just true or false.

Reply:
It shouldn't what are possible values for this? 0, 16, 4096?

Reply:
There might be different levels of vlan awareness, and those might differ between interfaces. (I'm not an expert.)

You now mentioning a different aspect, the maximum vlan id, also tells me that the variable probably is named to generic as well. I think there is valuable information to transport here, but we discuss a bit more in which form this could be stored.

Reply:
Thomas provided me with the list of Wiki values which are : Yes, No, 16, 4096. I just went with it. I'm happy to rename and convert it to a object or whatever complexity we want here.

Reply:
I updated it to True, False, 16 and 4096 for now.

Reply:
@adschm thoughts on this?

Reply:
Where the VLAN datafield comes from: Back in 2015 when we created the current ToH from the previous static tables, VLAN was mentioned there, so someone thought back then that it would be good to know. I have carried it with me for the past 5 years, always asking myself if it is really needed.

There was some discussion in the forum
https://forum.openwrt.org/t/maximum-number-of-vlans/34084
but I don't see a broad interest in this datafield.

Reply:
I think both the vlan-awareness and the available number of vlans is a valuable piece of information, and it's one that is not easy to extract from the code. It's also definitely related to OpenWrt "firmware". My struggle here is mainly how to translate it into properties in a reasonable way (i.e. having both true _and_ integers for the same field is odd IMO)